test/e2e/tests/ng-route-promise.spec.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+describe('ngRoute promises', function() {
+  beforeEach(function() {
+    loadFixture('ng-route-promise');
+  });
+
+  it('should wait for promises in resolve blocks', function() {
+    expect(element.all(by.tagName('li')).count()).toBe(5);
+  });
+
+  it('should time out if the promise takes long enough', function() {
+    // Don't try this at home kids, I'm a protractor dev
+    browser.manage().timeouts().setScriptTimeout(1100);
```

Comment:
@gkalpak oddly, this e2e test works fine event though the unit test fails.  I have a theory, which is that a *second* `waitForAngular()` is being introduced somewhere unexpected.  But I'm not sure where.